firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/cobs.txt

```diff
@@ -16,7 +16,11 @@ Scenario: The Util encode_cobs function correctly encodes buffers
 
   WHEN("The cobs::encode function is called on a buffer that contains these bytes '0x11 0x22 0x00 0x33'")
     THEN("The encode_cobs function reports ok status")
-    THEN("The encoded buffer is as expected '\x03\x11\x22\x02\x33'") {
+    THEN("The encoded buffer is as expected '\x03\x11\x22\x02\x33'")
+
+  WHEN("The cobs::encode function is called on a buffer that contains these bytes '0x11 0x22 0x00 0x33' and a non-empty output_buffer is passed")
```

Comment:
Regression coverage would be stronger if we had a test case where output_buffer needs to be resized up and a test case where output_buffer needs to be resized down, since we can't guarantee that the implementation will always handle both cases the same way. If you don't want to do this, let's at least specify the length of output_buffer (e.g. if we gave a non-empty output buffer of size 5 we might not catch any bugs preventing resizing, since the target size is 5 bytes anyways; so the test should specify that the length is not 5, or it should specify the exact length passed).

Reply:
the current case is resizing up as it's filled with 2-bytes , I'll add for down